main.cpp

```diff
@@ -35,19 +37,26 @@ void busy_thread()
 
     while(i--) {
         led1 = !led1;
-        wait_ms(wait_time);
+        thread_sleep_for(wait_time);
     }
 }
 
-void print_stats()
+void print_cpu_stats()
 {
     mbed_stats_cpu_t stats;
     mbed_stats_cpu_get(&stats);
 
+    //Calculate the CPU usage
+    uint64_t diff = (stats.idle_time - prev_idle_time);
+    uint8_t idle = (diff * 100) / (SAMPLE_TIME*1000);    // usec;
```

Comment:
It would be clearer to rename diff to `diff_usec` and change the comment to `percent`